ts/boot/migrateToPagination.ts

```diff
@@ -0,0 +1,66 @@
+import * as Either from "fp-ts/lib/Either";
+
+import {
+  messagesStatusSelector,
+  MessageStatus
+} from "../store/reducers/entities/messages/messagesStatus";
+import { Store } from "../store/actions/types";
+import { removeMessages } from "../store/actions/messages";
+
+type Failure = {
+  error: unknown;
+  messageId: string;
+};
+
+type MigrationResult = {
+  failed: Array<Failure>;
+  succeeded: Array<string>;
+};
+
+export default async function init(
+  store: Store,
+  upsert: (id: string, messageStatus: MessageStatus) => Promise<void>
+): Promise<MigrationResult> {
+  const data = messagesStatusSelector(store.getState());
+  const allIds = Object.keys(data);
+  if (allIds.length < 1) {
+    return { failed: [], succeeded: [] };
+  }
+  // TODO: dispatch to show the loader
+
+  const requests: Array<Promise<Either.Either<Failure, string>>> = allIds.map(
+    async id => {
+      const messageStatus = data[id];
+      if (messageStatus) {
+        try {
+          await upsert(id, messageStatus);
+          return Either.right<Failure, string>(id);
+        } catch (error) {
+          return Either.left<Failure, string>({ error, messageId: id });
+        }
+      } else {
+        return Either.right<Failure, string>(id);
+      }
+    }
+  );
+
+  return Promise.all(requests).then(results => {
+    const migrationResult: MigrationResult = { failed: [], succeeded: [] };
```

Comment:
Here it _could_ be possible to remove the ESLint comments using a `reduce` like:

```ts
const migrationResult: MigrationResult = results.reduce(
  (acc, next) =>
    next.fold(
      failure => ({ ...acc, failed: [...acc.failed, failure] }),
      id => ({ ...acc, succeeded: [...acc.succeeded, id] })
    ),
  {
    failed: [],
    succeeded: []
  } as MigrationResult
);
```

This is **computationally less performant** due to the spread operators (`...`) though. So if performances are an issue for the future, just ignore this 😆 (if we expect to migrate a **very big number** of messages). What do you think? 🙂

Reply:
I love it, and it was the first implementation as well 🤣  I ditched it because the imperative form seems easier to read. 
No strong opinions here, feel free to summon someone else if you believe you have a strong case for immutability.

Reply:
Same for the other comment, I can see your point, there are no advantages in using immutability here in my point of view, it would just remove the ESLint errors. I'm going to mark this as resolved!